packages/wallets/src/features/cashier/modules/SystemMaintenance/SystemMaintenanceContent.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Trans } from 'react-i18next';
+import { WalletText } from '../../../../components';
+
+type TSystemMaintenanceDescProps = {
+    currency?: string;
+    isCashierLocked?: boolean;
+    isCrypto?: boolean;
+    isDepositLocked?: boolean;
+    isWithdrawalLocked?: boolean;
+};
+
+const getSystemMaintenanceContent = ({
+    currency,
+    isCashierLocked,
+    isCrypto,
+    isDepositLocked,
+    isWithdrawalLocked,
+}: TSystemMaintenanceDescProps) => {
```

Comment:
for the content i suggest to not use conditional statements inside. Let it be all the strings and content and map it to a unique identifier. Once you want to use you can call it based on the condition you want to render it. This will remove the propnames aswelll and it will reduce the runtime to execute the logic 👍🏼

Reply:
The reason it is done this way is because we want to sort the lock conditions according to priority. So it'll go through all the conditions and if none is matched the children of the locked component would be rendered instead.

Mapping it would remove this priority order, unless if we map it in an array. But that would increase the runtime even more as we'll have to loop through the array instead. To cater to your suggestion and simplify the logic the component goes through all the if conditions and return the content only once at the end.